enhancements/monitoring/monitoring-config.md

```diff
@@ -0,0 +1,307 @@
+---
+title: monitoring-config
+authors:
+  - "@marioferh"
+  - "@danielmellado"
+  - "@jan--f"
+  - "@moadz"
+  - "@simonpasquier"
+reviewers:
+  - "@jan--f"
+  - "@moadz"
+  - "@simonpasquier"
+approvers:
+  - "@jan--f"
+  - "@moadz"
+  - "@simonpasquier"
+  - "@openshift/openshift-team-monitoring"
+api-approvers:
+  - "@deads2k"
+  - "@JoelSpeed"
+creation-date: 2024-04-26
+last-updated: 20204-26
+tracking-link:
+- "https://issues.redhat.com/browse/MON-1100"
+  
+---
+
+# CRD ClusterMonitoring
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+* Currently, the OCP monitoring stacks are configured using ConfigMaps (1 for platform monitoring and 1 for user-defined monitoring). In OpenShift though the best practice is to configure operators using custom resources.
+
+
+## Motivation
+
+* The specification is well known and to a degree self-documenting.
+* The Kubernetes API Server validates custom resources based on their API specification, so users get immediate feedback on errors instead of checking the ClusterOperator object and the cluster monitoring operator's logs. Common expressions language even allows to write complex validation logic, including cross-field tests.
+* Many users expect to interact with operators through a CRD.
+* Custom resources play better with GitOps workflows. 
+* CRDs supports multiple actors managing the same resource which is a key property for the Observability service of Advanced Cluster Management.
+
+### User Stories
+
+As a cluster administrator, I want to be able to add a CRD so that I can configure monitoring stack.
```

Comment:
IMO we can improve on the user story here. The end user doesn't really care about it being a CRD do they, what they want is a well structured and documented API with feedback when they misconfigure it, so that it is easy for them to configure the monitoring stack. The implementation of that I don't think they really care about.

Reply:
I'm not sure about it. This proposal focuses on the migration from ConfigMap to CR. We can add another field to explain its functionality or consider a separate proposal, the aim of this one is the migration. I believe it's important for the end user to understand what was possible before this feature and what can be achieved afterward

Reply:
The user story I don't think represents the reasons you're performing this migration. Something has spurred you to do this, feedback from users or something? I suspect the feedback is along the lines of my first comment in this thread

Users can apply configmaps, but there's no documented structure, nor is there any validation. By moving to a CRD, we get those things, but its those user facing improvements that the user story should focus on, not the technology

Reply:
I can add that to the User Story, but the reasons you wrote here, why we are migrating to CR are in the motivations. We can explain again what monitoring stack does but not sure if is the goal of this proposal


## Motivation

* The specification is well known and largely self-documenting.
* The Kubernetes API Server validates custom resources based on their API specification, so users get immediate feedback on errors instead of checking the ClusterOperator object and the cluster monitoring operator's logs. Common expressions language even allows to write complex validation logic, including cross-field tests.
* Many users expect to interact with operators through a CRD.
* Custom resources play better with GitOps workflows. 
* CRDs supports multiple actors managing the same resource which is a key property for the Observability service of Advanced Cluster Management.

Reply:
I agree with Joel, using a CRD is an impl outcome (the "what"), It'd be good to flesh out user stories driven by concrete value added for different personas (the "why we are investing in this"), i.e can you share a concrete example of how any of the statements in the "motivation" section impacts the cluster admin when setting up a monitoring stack?
e.g.
As a Service Provider I want to have granular ownership over some configuration knobs like remoteWrite so I can forward data to a centralized managed service without other entities clobbering my config.
As a Service Provider I want my managed (bad) configuration to fail on admission so I can propagate and signal the failure appropriately to watching components of the service and to a Service Consumer.

Reply:
Thanks, I need to think a little bit about what to add in user stories, thanks for the feedback.

Reply:
We had the request to improve a gitops style management for monitoring from various sides. Maybe that is worth mentioning here. Admins want to create a CR (possibly via gitops style systems) and observe the status of the created CR. Today this is spread over the config map and feedback via CMO status conditions and CMO logs.